### DISCUSSION TITLE
Viewer requesting retransmissions on packet loss?

### DISCUSSION THREAD

**Discussion by pahonen** [awslabs/amazon-kinesis-video-streams-webrtc-sdk-c#1902] [Q&A]
Can the viewer samples be configured to request retransmission on packet loss?

Master samples appear to do retransmission when RTCP_FEEDBACK_MESSAGE_TYPE_NACK is received from a different vendor WebRTC implementation.

**Comment by disa6302** [comment]
@pahonen ,

We currently do not support sending NACKs to other peers.

**Comment by BenOmerDev** [comment]
@pahonen, thank you for your question. If you wouldn't mind sharing the different vendor referenced, as well as media type and use case where retransmission is desired, that will help our team assess the demand for NACK sending support going forward.

**Reply by pahonen** [reply]
The use case is h264 video sent from master similar to kvsWebrtcClientMasterGstSample. The video is received by two types of viewers:
One similar to kvsWebRTCClientViewer and the other based on https://webrtc.googlesource.com/src/webrtc/.

Under network packet loss conditions, the first viewer based on this library will suffer from video quality degradation and is not emitting RTCP NACK to request retransmission for loss mitigation.
The second viewer based on that different library will emit RTCP NACK on detecting loss. The master based on this library even appears to be doing RTP retransmission on NACK reception.

In this use case, it would be better to retain video quality through retransmission in the face of reasonable network packet loss even if that would mean a bit of additional latency.
